lib-ee/emqx_ee_connector/src/emqx_ee_connector_influxdb.erl

```diff
@@ -348,30 +352,33 @@ do_async_query(InstId, Client, Points, ReplyFunAndArgs) ->
 %% -------------------------------------------------------------------------------------------------
 %% Tags & Fields Config Trans
 
-to_config(Lines) ->
-    to_config(Lines, []).
+to_config(Lines, Precision) ->
+    to_config(Lines, [], Precision).
 
-to_config([], Acc) ->
+to_config([], Acc, _Precision) ->
     lists:reverse(Acc);
-to_config(
-    [
-        #{
-            measurement := Measurement,
-            timestamp := Timestamp,
-            tags := Tags,
-            fields := Fields
-        }
-        | Rest
-    ],
-    Acc
-) ->
-    Res = #{
-        measurement => emqx_plugin_libs_rule:preproc_tmpl(Measurement),
-        timestamp => emqx_plugin_libs_rule:preproc_tmpl(Timestamp),
-        tags => to_kv_config(Tags),
-        fields => to_kv_config(Fields)
+to_config([Item0 | Rest], Acc, Precision) ->
+    Ts = maps:get(timestamp, Item0, undefined),
+    Item = #{
+        measurement => emqx_plugin_libs_rule:preproc_tmpl(maps:get(measurement, Item0)),
+        timestamp => preproc_tmpl_timestamp(Ts, Precision),
+        tags => to_kv_config(maps:get(tags, Item0)),
+        fields => to_kv_config(maps:get(fields, Item0))
     },
-    to_config(Rest, [Res | Acc]).
+    to_config(Rest, [Item | Acc], Precision).
+
+preproc_tmpl_timestamp(undefined, <<"ns">>) ->
```

Comment:
Not related to this PR in particular, but I noticed we do `atom_to_binary` on the precision.  :thinking: 

Do you know if there's a reason for that?

Reply:
I don't know either,  I guess it does `atom_to_binary()` for the influxdb to concatenate them into a query URL.

Reply:
Aha, I see!  It was just my own curiosity.  :smiley_cat: 

I think that the client should perform that internally, but I digress :joy_cat: